Move toward more standard tarball names, ship HTML documentation

nalimilan opened the pull request:
See https://github.com/JuliaLang/julia/issues/9977 and commit messages.

The first commit simply a adds new names to prepare the deprecation/transition; light-source-dist will eventually become dist. The second commit builds and installs HTML docs, but also ships them in the tarballs. It seems to work in my tests, but you should really check this is correct.

I'm not very happy about the copy/paste from light-source-dist to full-source-dist. Is there any simple solution in with Makefiles?

nalimilan committed:
Add alternative, more standard make target names

This will allow progressively moving to more standard names. #9977

nalimilan committed:
Build and install HTML documentation, include it in tarballs

This makes it possible to build the documentation without installing
a recent version of Python and without a network access required by
virtualenv. Remove the BUILDDIR variable for Sphinx, as its name is
too generic and would make the code more complex for no benefit.
Cf. #9977.

tkelman commented:
to fix the appveyor failure you'll need to cherry-pick b36966f15b2dfd9cf4304838d312b3624dcc692a, and maybe the other commits from #9554 as well

tkelman commented:
I committed the fixes to master and restarted the build here so now the docs can build on appveyor, but it takes about a minute per build. Can the `$(build_docdir): doc/_build/html` dependency be changed instead to `install: doc/_build/html` ?

nalimilan commented:
I've rebased and moved building the docs to the `install` target. Appears to work fine with both `light-source-dist` and `full-source-dist`.

tkelman commented:
I like it, though I do think `light-source-dist` and `full-source-dist` should reuse the common lines of code. Maybe something like

```
common-source-dist:
	<all common lines of code except the line that creates the tarball>

light-source-dist: common-source-dist
	<create tarball>

full-source-dist: common-source-dist
	cp light-source-dist.tmp full-source-dist.tmp
	<extra non-common steps for deps>
	<create tarball>
```

nalimilan committed:
Avoid full-source-dist failure if no .tar.xz file is present under deps/

Only LLVM 3.5 comes as a .tar.xz, which makes this target fail when
LLVM_VER=3.3 (the default).

nalimilan commented:
Good idea. Here's the new version.

I've also added a commit for a change I needed to run `make full-source-dist`. But I find it weird that nobody would have noticed it, so maybe I'm missing something.

tkelman commented:
Funny, github seems to have eaten my comments. Usually it keeps them around but collapses them. Anyway, the version as of 3b8050679122946341975cfa3a5f4d94101cf85e looks good to me, assuming you've checked that the tarballs from it look right. @staticfloat want to give this a once-over?

tkelman commented:
And I guess we should be able to backport some or all of this since this part doesn't change the meaning of any existing names yet.

nalimilan commented:
The tarballs look correct, at least the build succeeds and the HTML docs are installed as expected.

staticfloat commented:
Why the change from `$(BUILDDIR)` to `_build`?  Just curious.

nalimilan commented:
@staticfloat First I wanted to reuse `$(BUILDDIR)` in the top-level Makefile, but then I realized it would mean defining that variable in Make.inc, and its name sounded too general and possibly confusing. So I thought about renaming it to `$(SPHINX_BUILDDIR)`, but eventually wondered about the use case for setting that variable. We don't support different build directories for anything else. But if you like I can change it.

staticfloat commented:
I don't mind.  Just curious.  I have no further critique.

tkelman commented:
I guess we need to update the readme now that python is a requirement for `make install`. I thought it would be necessary for LLVM, but apparently not...

tkelman commented:
I backported just the first commit in 0d1e7bcf261e4bad1587d55c929b7647b41215b9, we can skip the others.
